Tema2.js

```diff
@@ -0,0 +1,102 @@
+//problem 1
+console.log(problem1(3));
+function problem1(number) {
+    if (number % 2 != 0)
+        return number * number;
+    else
+        return number - 2;
+}
+
+
+//problem 2
+const savePrimes = primes => {
+    //console.log(savePrimes[1]);
+    const arrayPrimes = new Array;
+    let dimension = primes.length;
+    let contor = 0;
+
+    for (let i = 0; i < dimension; i++) {
+        let prim = true;
+        if (primes[i] < 2)
+            continue;
+        for (let div = 2; div < primes[i]; div++) {
+            if (primes[i] % div == 0)
+                prim = false;
+
+        }
+        if (prim) {
+            arrayPrimes[contor++] = primes[i];
+        }
+    }
+    console.log(arrayPrimes);
+}
+
+const arr = [1, 2, 3, 4, 5, 6, 7, 8, 9];
+savePrimes(arr);
+
+
+//problem 3 (IIFE)
+const saveStrings = [1, '1', 'salut', 2, 6, true, 'true'];
+((arrStrings) => {
+    const justStrings = new Array;
+    let dim = arrStrings.length;
+    let k = 0;
+
+        arrStrings.forEach(item => {
```

Comment:
Linea asta se poate rezuma la:
    arrStrings.forEach(item => typeof(item) === 'string' && justStrings.push(item));
Din ea reies cateva chestii:
- folosirea .push in loc de un contor (ca si in ex anterior)
- sintaxa (conditie1 && conditie2) face lucrul urmator: verifica daca "conditie1" e adevarata (sau truthy) iar apoi verifica "conditie2", in cazul de fata verifica daca item-ul e string, iar daca e adevarat, verifica (executa) "justStrings.push(item)" si pune item-ul in array

Reply:
De asemenea, _dim_ este in plus acolo